controllers/toggle_components.go

```diff
@@ -90,21 +90,23 @@ func (r *MultiClusterEngineReconciler) ensureNoConsoleMCE(ctx context.Context, b
 		}
 		return ctrl.Result{RequeueAfter: requeuePeriod}, nil
 	}
-
 	r.StatusManager.RemoveComponent(toggle.EnabledStatus(namespacedName))
-	r.StatusManager.AddComponent(toggle.DisabledStatus(namespacedName, []*unstructured.Unstructured{}))
 	if !ocpConsole {
 		r.StatusManager.AddComponent(status.ConsoleUnavailableStatus{
 			NamespacedName: types.NamespacedName{Name: "console-mce-console", Namespace: backplaneConfig.Spec.TargetNamespace},
 		})
+	} else {
+		r.StatusManager.AddComponent(toggle.DisabledStatus(namespacedName, []*unstructured.Unstructured{}))
 	}
 
 	// Deletes all templates
 	for _, template := range templates {
-		result, err := r.deleteTemplate(ctx, backplaneConfig, template)
-		if err != nil {
-			log.Error(err, fmt.Sprintf("Failed to delete Console MCE template: %s", template.GetName()))
-			return result, err
+		if ocpConsole && !strings.Contains(strings.ToLower(template.GetAPIVersion()), "console") {
```

Comment:
This causes resources in the console.openshift.io apigroup to not be cleaned up when console is disabled